pkg/k8shandler/visualization.go

```diff
@@ -40,46 +43,30 @@ var (
 // CreateOrUpdateVisualization reconciles visualization component for cluster logging
 func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateVisualization(proxyConfig *configv1.Proxy) (err error) {
 	if clusterRequest.cluster.Spec.Visualization == nil || clusterRequest.cluster.Spec.Visualization.Type == "" {
-		if err = clusterRequest.removeKibana(); err != nil {
-			return
+		if err := clusterRequest.removeKibana(); err != nil {
+			logrus.Error(err.Error())
+			return nil
 		}
 		return nil
 	}
-	if clusterRequest.cluster.Spec.Visualization.Type == logging.VisualizationTypeKibana {
-
-		if err = clusterRequest.CreateOrUpdateServiceAccount(kibanaServiceAccountName, &kibanaServiceAccountAnnotations); err != nil {
-			return
-		}
-
-		if err = clusterRequest.RemoveClusterRoleBinding("kibana-proxy-oauth-delegator"); err != nil {
-			return
-		}
-
-		if err = clusterRequest.createOrUpdateKibanaService(); err != nil {
-			return
-		}
-
-		if err = clusterRequest.createOrUpdateKibanaRoute(); err != nil {
-			return
-		}
-
-		if err = clusterRequest.createOrUpdateKibanaSecret(); err != nil {
-			return
-		}
-
-		if err = clusterRequest.createOrUpdateKibanaConsoleExternalLogLink(); err != nil {
-			return
-		}
+	if err = clusterRequest.removeOldKibana(); err != nil {
+		logrus.Error(err.Error())
+		return nil
+	}
 
-		if err = clusterRequest.RemoveOAuthClient("kibana-proxy"); err != nil {
-			return
-		}
+	if err = clusterRequest.createKibanaCR(); err != nil {
+		logrus.Error(err.Error())
+		return nil
+	}
 
-		if err = clusterRequest.createOrUpdateKibanaDeployment(proxyConfig); err != nil {
-			return
-		}
+	if err = clusterRequest.createOrUpdateKibanaSecret(); err != nil {
+		logrus.Error(err.Error())
+		return nil
+	}
 
-		clusterRequest.UpdateKibanaStatus()
+	if err = clusterRequest.UpdateKibanaStatus(); err != nil {
+		logrus.Error(err.Error())
+		return nil
```

Comment:
nit but this `return nil` is unnecessary

Reply:
`return nil` removed